src/main/java/randoop/sequence/SequenceCollection.java

```diff
@@ -50,6 +53,13 @@ public class SequenceCollection {
   /** Number of sequences in the collection: sum of sizes of all values in sequenceMap. */
   private int sequenceCount = 0;
 
+  /**
+   * Determine if demand-driven input creation can be used. This only indicates if it can be used,
+   * not if it should be used. Demand-driven input creation is disabled for getSequencesForType()
+   * when it is called from within demand-driven input creation to avoid infinite recursion.
```

Comment:
This seems undesirable.  Does it mean that demand-driven is only one level deep?

Could you pass a "level" variable in, and if it is (say) 5 or more then give up?

Reply:
I think demand-driven as a whole is only meant to be called once per missing type. The "depth" of the search is achieved internally by demand-driven's iterativeProducerMethodSearch. It searches until it finds all the methods that create the required type `t` starting from exploring all reachable methods from the formal parameters of constructors and methods that return `t`. Please correct me if you disagree.